### DISCUSSION TITLE
Improve Entity Ruler accuracy

### DISCUSSION THREAD

**Discussion by aoleiReiz** [explosion/spaCy#7711] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour

`nlp = spacy.load(model_dir)`
`nlp.add_pipe(nlp.create_pipe('sentencizer'))`
`ruler = EntityRuler(nlp)`
`patterns = generate_patterns_from_file()`
`ruler.add_patterns(patterns)`
`nlp.add_pipe(ruler)`

the model failed to extract most of the rule patterns, for example:
the model can identify "IKEA" as label "EXT-ORG", but after add the rule based pattern
 `{"label": "ORG-EXT", "pattern": [{"LOWER": "ikea ab"}]}` 
the model failed to extract it in text "ikea ab data lake"

## Your Environment

* Operating System: Windows 10
* Python Version Used: 3.6.8
* spaCy Version Used: 2.3.5
* Environment Information:

**Comment by adrianeboyd** [comment, marked as answer]
Hi, please have a look at the docs about token patterns, in particular the warning in the red box: https://spacy.io/usage/rule-based-matching#adding-patterns

You may want to use phrase patterns instead of token patterns with the entity ruler, see: https://spacy.io/usage/rule-based-matching#entityruler-patterns

**Reply by aoleiReiz** [reply]
thanks for quick replying!
does phrase pattern works ignoring lower and upper case, I see the  example pattern is 'MyCorp Inc.',  will it work for "mycorp inc", thanks

**Reply by adrianeboyd** [reply]
You can set a single attribute to use for the phrase matcher patterns: https://spacy.io/api/entityruler#config

**Reply by aoleiReiz** [reply]
@adrianeboyd Thanks, after adding phrase pattern, it can identity most of the ents, thanks for your help
